src/test_l3_dsDisplay.c

```diff
@@ -0,0 +1,435 @@
+
+    dsError_t status   = dsERR_NONE;
+
+    UT_LOG_INFO("Calling dsDisplayTerm()");
+    status = dsDisplayTerm();
+    UT_LOG_INFO("Result dsDisplayTerm() dsError_t=[%s]", UT_Control_GetMapString(errorMappingTable, status));
+    DS_ASSERT(status == dsERR_NONE);
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+ * @brief Retrieves the display handle based on user-selected video port.
+ *
+ * Allows the user to select a video port and retrieves the corresponding display handle.
+ *
+ * **Test Group ID:** 03@n
+ * **Test Case ID:** 003@n
+ *
+ * **Test Procedure:**
+ * Refer to Test specification documentation
+ * [ds-display_L3_Low-Level_TestSpecification.md](../../docs/pages/ds-display_L3_Low-Level_TestSpecification.md)
+ */
+static void dsGetDisplay_getHandle()
+{
+    gTestID = 3;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status   = dsERR_NONE;
+    int32_t choice,port;
+
+    UT_LOG_INFO(" \t  Supported Video Port are:");
+    UT_LOG_INFO("------------------------------------------");
+    for (port = 0; port < gDSvideoPort_NumberOfPorts; port++) {
+        UT_LOG_INFO("\t%d.  %-20s\n", port+1, UT_Control_GetMapString(videoPortTypeMappingTable, gDSVideoPortConfiguration[port].typeid));
+    }
+    UT_LOG_INFO("------------------------------------------");
+    UT_LOG_INFO(" Select the Video Port:");
+    scanf("%d", &choice);
+    port = choice - 1;
+    UT_LOG_INFO("Port %d choice %d ",port,choice);
+    if(choice < 1 || choice > gDSvideoPort_NumberOfPorts) {
+        UT_LOG_ERROR("Invalid Port choice\n");
+        return;
+    }
+
+    UT_LOG_INFO("Calling dsGetDisplay(IN:type:[%s],IN:index:[%d])", UT_Control_GetMapString(videoPortTypeMappingTable, gDSVideoPortConfiguration[port].typeid));
+    status = dsGetDisplay(gDSVideoPortConfiguration[port].typeid, gDSVideoPortConfiguration[port].index, &displayHandle);
+
+    UT_LOG_INFO("Result dsGetDisplay(Handle:[0x%0X]]) dsError_t=[%s]", displayHandle, UT_Control_GetMapString(errorMappingTable, status));
+    DS_ASSERT(status == dsERR_NONE);
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+
+}
+
+/**
+ * @brief Retrieves Extended Display Identification Data (EDID) for the display.
+ *
+ * Retrieves the EDID information for the display associated with the selected handle.
+ *
+ * **Test Group ID:** 03@n
+ * **Test Case ID:** 004@n
+ *
+ * **Test Procedure:**
+ * Refer to Test specification documentation
+ * [ds-display_L3_Low-Level_TestSpecification.md](../../docs/pages/ds-display_L3_Low-Level_TestSpecification.md)
+ */
+void dsDisplay_GetEDID() 
+{
+    gTestID = 4;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    dsDisplayEDID_t *edid =NULL;
+
+    dsGetDisplay_getHandle();
+
+    if (displayHandle == -1) {
+        UT_LOG_ERROR("Failed to get display handle\n");
+        return;
+    }
+
+    edid = (dsDisplayEDID_t *)malloc(sizeof(dsDisplayEDID_t));
```

Comment:
use a static variable instead of malloc to make code simple. you can avoid free in error cases

Reply:
Done